shoop/front/apps/auth/forms.py

```diff
@@ -30,6 +30,11 @@ def clean_username(self):
         if '@' in username:
             try:
                 username = user_model.objects.get(email=username).username
+            except ObjectDoesNotExist:
+                raise forms.ValidationError(
+                    _("Email and password didn't match, please try again."),
```

Comment:
That would still be vulnerable to timing attacks, since then the password hashing would not be ran. See https://code.djangoproject.com/ticket/20760 for details.

IMHO it's probaly just best to just remove the `if '@' in username` part and always first query for an user by username and then query by email (even if exists with username to make timing attacks harder) and then select the first one that exists. I.e. something like this:
```python
# Note: Always searches by username AND by email from database, to prevent timing attacks
try:
    user_by_name = user_model._default_manager.get_by_natural_key(username)
except user_model.DoesNotExist:
    user_by_name = None
try:
    user_by_email = user_model._default_manager.get(email=email)
except user_model.DoesNotExist:
    user_by_email = None
except MultipleObjectsReturned:
    raise forms.ValidationError(...)  # or should we mask this too for security? That's against the spec though

if not user_by_name and user_by_email:
    return getattr(user_by_email, user_model.USERNAME_FIELD)
return username
```